From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Leung <ianleung7@gmail.com>
Date: Mon, 2 Feb 2026 18:01:25 +0800
Subject: rk3566-orangepi-3b: Enable boot to sata

This enables the kernel to read from an attached SATA M.2 SSD, allowing
booting to SSD.

Signed-off-by: Ian Leung <ianleung7@gmail.com>
---
 arch/arm64/boot/dts/rockchip/rk3566-orangepi-3b.dtsi | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-orangepi-3b.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-orangepi-3b.dtsi
index d539570f5..6a4ab7db9 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-orangepi-3b.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-orangepi-3b.dtsi
@@ -51,10 +51,12 @@ vcc3v3_pcie30: regulator-3v3-vcc-pcie30 {
 		enable-active-high;
 		gpios = <&gpio0 RK_PB7 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pcie20_pwren>;
 		regulator-name = "vcc3v3_pcie30";
+		regulator-always-on;
+		regulator-boot-on;
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
 		vin-supply = <&vcc3v3_sys>;
 	};
 
@@ -452,11 +454,11 @@ &i2s1m0_sdi0
 &pcie2x1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pcie20_pins>;
 	reset-gpios = <&gpio0 RK_PB6 GPIO_ACTIVE_HIGH>;
 	vpcie3v3-supply = <&vcc3v3_pcie30>;
-	status = "okay";
+	status = "disabled";
 };
 
 &pinctrl {
 	bluetooth {
 		bt_reg_on_h: bt-reg-on-h {
@@ -533,10 +535,15 @@ &pmu_io_domains {
 &saradc {
 	vref-supply = <&vcca_1v8>;
 	status = "okay";
 };
 
+&sata2 {
+	target-supply = <&vcc3v3_pcie30>;
+	status = "okay";
+};
+
 &sdhci {
 	bus-width = <8>;
 	cap-mmc-highspeed;
 	max-frequency = <200000000>;
 	mmc-hs200-1_8v;
-- 
Created with Armbian build tools https://github.com/armbian/build

